### PR TITLE
fix: add tests for 'table' and 'columnName' usage in model definitions

### PR DESCRIPTION
**Pull request by emonddr**
This PR is providing the test cases for  a fix in loopback-ibmdb/lib/ibmdb.js' _replace() method (See [PR #80](https://github.com/strongloop/loopback-ibmdb/pull/80)  ).

The test cases should exercise loopback-datasource-juggler/lib/dao.js' `replaceById()` function so that it calls loopback-ibmdb/lib/ibmdb.js' `_replace()` function.

We should have : 

 - one test case that **does** use `table` and `columnName` values in the model and property definitions
 - one test case that **does not** use the `table` and `columnName` values in the model and property definitions
- each test case should perform a `create`, followed by a `replaceById`

These test cases cannot be placed in the 
`https://github.com/strongloop/loopback-next/tree/master/packages/repository-tests` test suites
since these test cases are generic and don't specify connector-specific overrides like:

```
"dashdb" : {
    "table" : "AWARD"
}
```

or

```
"dashdb" : {
    "columnName" : "AWARD_ID"
}
```

## Related Issue

connect to: https://github.com/strongloop/loopback-ibmdb/issues/79 

Dependent on PR https://github.com/strongloop/loopback-ibmdb/pull/80

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-connector-dashdb) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

File: test/dashdb.replaceById.test.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          function(err, updatedBook) {
+            if (err) return done(err);
+            assert(updatedBook);
+            assert.equal(updatedBook.id, id);
+            assert.equal(updatedBook.title, REVISED_BOOK_TITLE);
+            done();
+          });
+      });
+    });
+  });
```
Comment (bajtos) [inline test/dashdb.replaceById.test.js]: We are trying to move away from callback-based tests and use async/await wherever possible, unfortunately the effort to migrate the exist code base is large and therefore we haven't made the change yet. Let's not build up this technical debt by creating more callback-based code.

Can you please rewrite this test to async/await style?

```js
it('create and replace a book', async () => {
  await db.automigrate('Book');
  const newBook = await Book.create({title: BOOK_TITLE});
  assert(newBook);
  // and so on
});
```

The same comment applies to other newly added tests too.

File: package.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
     "async": "^3.1.0",
     "debug": "^4.1.1",
     "loopback-connector": "^4.0.0",
-    "loopback-ibmdb": "^2.3.0",
-    "strong-globalize": "^5.0.0"
+    "strong-globalize": "^5.0.0",
+    "loopback-ibmdb": "github:strongloop/loopback-ibmdb#dremond_fix_replace"
```
Comment (dhmlau) [inline package.json]: Please remember to change it to the released version once your PR in loopback-ibmdb has landed.

File: test/dashdb.replaceById.test.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+const BOOK_TITLE = 'Rocky I';
+const REVISED_BOOK_TITLE = 'Rocky II';
+
+/*
+   This test suite is to test the changes surrounding the fix
+   in loopback-ibmdb related to issue :
+   https://github.com/strongloop/loopback-ibmdb/issues/79
+*/
+
+describe('replaceById with BOOK_ID', function() {
```
Comment (jannyHou) [inline test/dashdb.replaceById.test.js]: If people don't know the background of this issue, I feel it's hard for them to understand what is the difference of "BOOK_ID" and "book_id" and why there are 2 tests.

Maybe rephrase the test title to be:
'replaceById with custom column name for id'

File: package.json (hunk 2 above)
Comment (emonddr) [reply package.json]: @dhmlau , I updated the version for loopback-ibmdb to `^2.6.0` this morning. We released 2.6.0 to npmjs yesterday.
